kip-0030.md

```diff
@@ -0,0 +1,227 @@
+
+```ts
+type ConnectEventPayload = {
+  source: "kadena-spirekey";
+  name: "connected";
+  payload: Account;
+};
+```
+
+If the user cancels the connection process, the wallet should close the window
+and emit an event with the following information:
+
+```ts
+type CanceledConnectEventPayload = {
+  source: "kadena-spirekey";
+  name: "canceled:connect";
+  payload?: void;
+};
+```
+
+## Sign
+
+The application must provide a button to sign a transaction. The SpireKey SDK
+displays a notification when another window has activity. This notification
+allows the wallet to open a new window and ask the user to sign the transaction.
+The wallet should open on the path using the `/sign` endpoint.
+
+The wallet is responsible for displaying the transaction to the user and asking
+for the user's approval. How the user is guided through this process is up to
+the wallet.
+
+The application should send a list of transactions to the wallet and,
+optionally, a list of accounts that need to have fungibles avaible for the
+transaction. The wallet can use this information to help the user send the funds
+to the requested chain.
+
+The application should send this information to the wallet in the form of hash
+parameters. The parameters are: `transactions` and `accounts`.
+
+- The `transactions` parameter is expected to be a JSON encoded array of
+  transactions.
+- The `accounts` parameter is expected to be a JSON encoded array of
+  `OptimalTransactionsAccount` which is a subset of an `Account`.
+
+After the user has signed for all necessary transactions, the wallet should
+submit any transactions that are needed to send the funds to the requested
+chain. The TransactionDescriptors for those transactions should be included in
+the `postMessage` event.
+
+The Wallet should emit the `postMessage` event on the `window` of the wallet
+with the following information:
+
+```ts
+type SignEventPayload = {
+  source: "kadena-spirekey";
+  name: "signed";
+  payload: {
+    accounts: Account[];
+    txs: Record<string, { sig: string; pubKey: string }[]>;
+  };
+};
+```
+
+If the user cancels the signing process, the wallet should close the window and
+emit an event with the following information:
+
+```ts
+type CanceledSignEventPayload = {
+  source: "kadena-spirekey";
+  name: "canceled:sign";
+  payload?: void;
+};
+```
+
+## Type references
+
+### ConnectEventPayload
+
+```ts
+type ConnectEventPayload = {
+  source: "kadena-spirekey";
+  name: "connected";
+  payload: Account;
+};
+```
+
+### Account
+
+```ts
+import type { ChainId, ITransactionDescriptor } from "@kadena/client";
+
+export type QueuedTx = ITransactionDescriptor;
+
+export type OptimalTransactionsAccount = Pick<
+  Account,
+  "chainIds" | "accountName" | "networkId" | "requestedFungibles"
+>;
+export type Account = {
+  alias: string;
+  accountName: string;
+  minApprovals: number;
+  minRegistrationApprovals: number;
+  balance: string;
+  devices: Device[];
+  guard?: Guard;
+  keyset?: Keyset;
```

Comment:
Is this guard significantly different from the keyset? It would be good to explain the difference.

Reply:
I'll update the supporting text around it. :+1: